rust/blockstore/src/arrow/blockfile.rs

```diff
@@ -102,11 +102,54 @@ impl ArrowBlockfileWriter {
                 removed = self.root.sparse_index.remove_block(&delta.id);
             }
             if !removed {
+                self.root
+                    .sparse_index
+                    .set_count(delta.id, delta.len() as u32)
+                    .map_err(|e| Box::new(e) as Box<dyn ChromaError>)?;
                 let block = self.block_manager.commit::<K, V>(delta);
                 blocks.push(block);
             }
         }
 
+        // MIGRATION(10/15/2024 @hammadb) Get all the blocks and manually update the sparse index
+        if self.root.version == Version::V1 {
+            self.root.version = Version::V1_1;
+            let block_ids;
+            // Guard the sparse index data access with a lock
+            // otherwise we have to hold the lock across an await
+            {
+                let sparse_index_data = self.root.sparse_index.data.lock();
+                block_ids = sparse_index_data
+                    .forward
+                    .values()
+                    .copied()
+                    .collect::<Vec<Uuid>>();
+            }
+            for block_id in block_ids.iter() {
+                let block = self.block_manager.get(block_id).await;
```

Comment:
I deliberately don't do a fast preload to prioritize system-wide concurrency

Reply:
makes sense